### PR TITLE
Refactor Fuzz implementation

### PR DESCRIPTION
**Pull request by pjbgf**
Following-up from the security audit, I reviewed [the initial Fuzz PR](https://github.com/fluxcd/kustomize-controller/pull/434) and created this PR that supersedes it.

Main changes:
- Structure the fuzz implementation to be closer to what fuzz testing might look like once leveraging the [golang 1.18 built-in fuzz feature](https://go.dev/doc/fuzz/).
- Add Makefile target to enable smoke testing fuzzers locally.
- Add github workflow to smoke test fuzzers.
- Add more fuzz targets for age and pgp.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,7 +22,7 @@ ENVTEST_ARCH ?= amd64
 all: manager
 
 # Download the envtest binaries to testbin
-ENVTEST_ASSETS_DIR=$(shell pwd)/testbin
+ENVTEST_ASSETS_DIR=$(shell pwd)/build/testbin
```
Comment (hiddeco) [inline Makefile:25]: Is `testbin/` still used for something (as it's still included in `.gitignore`)

File: controllers/suite_test.go [hunk 2]
```diff
@@ -152,6 +139,30 @@ func TestMain(m *testing.M) {
...
+		testEventsH = controller.MakeEvents(testEnv, controllerName, nil)
+		testMetricsH = controller.MustMakeMetrics(testEnv)
+		reconciler := &KustomizationReconciler{
+			ControllerName:  controllerName,
+			Client:          testEnv,
+			EventRecorder:   testEventsH.EventRecorder,
+			MetricsRecorder: testMetricsH.MetricsRecorder,
+		}
+		if err := (reconciler).SetupWithManager(testEnv, KustomizationReconcilerOptions{MaxConcurrentReconciles: 4}); err != nil {
+			panic(fmt.Sprintf("Failed to start GitRepositoryReconciler: %v", err))
```
[the remark this answers is not in the document]
Comment (pjbgf) [reply controllers/suite_test.go]: Nice catch. :bow:

File: Makefile (hunk 1 above)
Comment (pjbgf) [reply Makefile:25]: Not at all, I have pushed a new version removing it from the `.gitignore`. Thanks for pointing it out.

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @pjbgf

**Pull request merged**